payment/payment.go

```diff
@@ -95,7 +125,20 @@ func FromScript(script []byte, net *network.Network) (*Payment, error) {
 	} else {
 		tmpNet = net
 	}
-	redeem := &Payment{Network: tmpNet, Script: script}
+
+	scriptHash := make([]byte, 0)
+	if script[0] == txscript.OP_0 {
+		scriptHash = append(scriptHash, script[2:]...)
+	}
+	if script[0] == txscript.OP_HASH160 {
+		scriptHash = append(scriptHash, script[2:len(script)-1]...)
+	}
+	if script[len(script)-1] == txscript.OP_CHECKMULTISIG {
+		scriptHash = hash160(script)
+	}
+
+	redeem := &Payment{Network: tmpNet, Hash: scriptHash, Script: script,
+		BlindingKey: blindingKey}
 	return FromPayment(redeem)
```

Comment:
here we should not call the FromPayment but rather returning the payment instance:
```suggestion
	return &Payment{
	   Network: tmpNet,
	   Hash: scriptHash,
	   Script: script,
	   BlindingKey: blindingKey,
	}, nil
```

Reply:
Done